### PR TITLE
[WIP] terminal: fix string width issues

### PR DESCRIPTION
**Pull request by erw7**
Change libvterm to use neovim functions instead of `vterm_unicode_width()` and `vterm_unicode_is_combining()`.

This is the change equivalent to v8.0.0985(https://github.com/vim/vim/commit/6d0826dfbba9880820d9ec221327e4250bbf6540) in vim.

### CODE REVIEW

**Review by jamessan — COMMENTED** [review]
This doesn't help people packaging neovim, since they don't used the bundled dependencies.  @leonerd, can you please review and determine how to address this in libvterm upstream?

File: src/nvim/mbyte.c [hunk 1]
```diff
@@ -458,6 +458,19 @@ static bool intable(const struct interval *table, size_t n_items, int c)
   return false;
 }
 
+#ifdef NVIM_VTERM_HAS_USER_FUNC
+///
+/// utf_char2cells() with different argument type for libvterm.
```
Comment (bfredl) [inline src/nvim/mbyte.c:463]: this sounds a little bit too much of a lie (as it surely does more than call `utf_char2cells`)

**Comment by leonerd** [comment]
Yah this wants to be an upstream libvterm change. I'm unconvinced of that precise API shape for doing it,

Best would be to discuss it as a libvterm thing first and come back to nvim later when that's done.

**Comment by erw7** [comment]
@leonerd Where can we have a discussion about libvterm? Is this the Answers of launchpad?

See #13036 for why this is necessary. I don't care what form the API is in, but I would like a way to enforce the character width on libvterm. Ideally all applications would use `wcwidth` and `wcwidth` would return a font-based character width, but the reality is that this is not the case and I think something like this is needed.

**Pull request closed without merging**